artichoke-backend/src/sys/protect.rs

```diff
@@ -164,20 +164,28 @@ struct IsRange {
 
 impl Protect for IsRange {
     unsafe extern "C" fn run(mrb: *mut sys::mrb_state, data: sys::mrb_value) -> sys::mrb_value {
+        use sys::mrb_range_beg_len::{MRB_RANGE_OK, MRB_RANGE_OUT, MRB_RANGE_TYPE_MISMATCH};
+
         let ptr = sys::mrb_sys_cptr_ptr(data);
         let Self { value, len } = *Box::from_raw(ptr.cast::<Self>());
         let mut start = mem::MaybeUninit::<sys::mrb_int>::uninit();
         let mut range_len = mem::MaybeUninit::<sys::mrb_int>::uninit();
         let check_range = sys::mrb_range_beg_len(mrb, value, start.as_mut_ptr(), range_len.as_mut_ptr(), len, false);
-        if check_range == sys::mrb_range_beg_len::MRB_RANGE_OK {
-            let start = start.assume_init();
-            let range_len = range_len.assume_init();
-            let out = Range { start, len: range_len };
-            let out = Box::new(out);
-            let out = Box::into_raw(out);
-            sys::mrb_sys_cptr_value(mrb, out.cast::<c_void>())
-        } else {
-            sys::mrb_sys_nil_value()
+        match check_range {
```

Comment:
Am I interpreting this correctly, does mruby change a Range into a start + len for us? That's pretty nice.

e.g.
```
a = "hello"
a.byteslice(-3..-3)
```

In the above, `sys::mrb_range_beg_len` would return `start: 2, len: 1`? Very nice if it does

Reply:
yes that's correct. The routine looks like this:

https://github.com/artichoke/artichoke/blob/7bf3f033e020dd60f4b5fd5974dcb19af2617f6d/artichoke-backend/vendor/mruby/src/range.c#L456-L488